### ISSUE TITLE
build_web_compilers + DDC: inject warning if browser != chrome

### ISSUE THREAD

**Issue by kevmoo** [dart-lang/build#2233]
If you're not using chrome when hitting a DDC-compiled app, you'll likely have a bad time.

Would be good to add some sort of warning to help folks who hit this and think that something is broken.

@efortuna hit this

CC @vsmenon

**Comment by vsmenon** [comment]
I think this is the same as:

https://github.com/dart-lang/webdev/issues/318

@grouma

**Comment by kevmoo** [comment]
Better to do it here – so even if folks aren't using webdev, they see it – right?

**Comment by vsmenon** [comment]
I believe the relevant logic was moved to webdev.

Note, I think this is only an issue if you pass `--debug` to webdev.  Otherwise, gallery (e.g.) loads fine in Safari / FF.

**Comment by kevmoo** [comment]
But DDC isn't tested AT all on other browsers, right? I want to set
expectations so we don't break folks...

**Comment by vsmenon** [comment]
Yes, it's not tested / officially supported on other browsers.  DevTools also won't work (it's based on Chrome Debug Protocol).  

Might make sense to report that and continue.  (I.e., not sure it makes sense to *not* try loading the app.)

**Comment by kevmoo** [comment]
Yeah, I was thinking of something like a bubble at the bottom of the screen
or something. Ship it as part of the bootstrap JS in build_web_compilers or
something.

**Comment by vsmenon** [comment]
Here's a first cut:

https://github.com/dart-lang/webdev/pull/357

**Comment by jakemac53** [comment]
We did fix the devtools issue here - @kevmoo did we want to do a more general thing? I am not so sure given that ddc does generally work.

**Comment by kevmoo** [comment]
Happy to hold off – unless we get a lot of bug reports. Sound reasonable?

**Comment by jakemac53** [comment]
Sure, can we close the issue (if we are waiting for new issues anyways, we can just use those)

**Comment by davidmorgan** [comment]
Closing as stale.
